apps/frontend/src/pages/tap.tsx

```diff
@@ -1,30 +1,35 @@
 import { useEffect } from "react";
 import { useRouter } from "next/router";
-import { TapParams, ChipTapResponse, errorToString, ChipIssuer } from "@types";
+import { ChipTapResponse, errorToString, ChipIssuer } from "@types";
 import { toast } from "sonner";
 import { storage } from "@/lib/storage";
 import {
   registerChip,
   tapChip,
   updateTapLeaderboardEntry,
-  updateWorkoutLeaderboardEntry,
+  updateLannaWorkoutLeaderboardEntry,
 } from "@/lib/chip";
 import { CursiveLogo } from "@/components/ui/HeaderCover";
 import { logClientEvent } from "@/lib/frontend/metrics";
 import { SupportToast } from "@/components/ui/SupportToast";
 import { ERROR_SUPPORT_CONTACT } from "@/constants";
 import { shareableUserDataToJson } from "@/lib/user";
+import { hasRecentAddChipRequest } from "@/lib/chip/addChip";
 
 const TapPage: React.FC = () => {
   const router = useRouter();
 
   useEffect(() => {
     const handleTap = async () => {
       logClientEvent("tap-chip", {});
-      const { chipId } = router.query;
-
-      if (!chipId || typeof chipId !== "string") {
-        logClientEvent("tap-chip-invalid-chip-id", {});
+      const tapParams = Object.fromEntries(
+        Object.entries(router.query)
+          .filter(([, value]) => typeof value === "string")
+          .map(([key, value]) => [key, value as string])
+      );
```

Comment:
Does this generalize over both chip types?

Reply:
yup